### PR TITLE
Awesome books es6

### PR DESCRIPTION
**Pull request by MargaretTechworld**
The following was added to this repository:
- Set up my project:
- Recreated my Awesome Books app following the rules
- Displayed the current date below the navigation bar using Luxon library 
- Added a discriptive README file.

Thanks for your review

### CODE REVIEW

File: modules/bookcollection.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import Book from './bookclass.js';
+
+export default class BookCollection {
+  constructor() {
+    this.collection = JSON.parse(localStorage.getItem('books')) || [];
+    this.buttonAdd = document.querySelector('.add-button');
+    this.bookTitle = document.querySelector('.input-field1');
+    this.bookAuthor = document.querySelector('.input-field2');
+    this.bookShelf = document.querySelector('.books-section');
+    // this.removeButton = document.querySelectorAll('remove-button')
```
Comment (levy002) [inline modules/bookcollection.js]: - [ ] Kindly make sure to remove all comments in order to keep your codebase clean.

File: modules/bookcollection.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    this.bookShelf.innerHTML = displayBooks.join('');
+
+    const removeButtons = document.querySelectorAll('.remove-button');
+    removeButtons.forEach((btn) => {
+      btn.addEventListener('click', () => {
+        const { index } = btn.dataset;
+        this.removeBook(index);
+      });
+    });
+  }
```
Comment (levy002) [inline modules/bookcollection.js]: - [ ] Kindly ese arrow functions in class methods for improved `this` binding. Enhances code clarity and maintainability.
```suggestion
   addBook = () => {
    if (this.bookTitle.value !== '' && this.bookAuthor.value !== '') {
      const book = new Book(this.bookTitle.value, this.bookAuthor.value);
      this.collection.push(book);
    }
  }

  removeBook = (index) => {
    this.collection.splice(index, 1);
    this.saveBooks();
    this.showBooks();
  }

  saveBooks = () => {
    localStorage.setItem('books', JSON.stringify(this.collection));
  }

  clearInput = () => {
    this.bookTitle.value = '';
    this.bookAuthor.value = '';
  }

  showBooks = () => {
    const displayBooks = this.collection.map((book, index) => `<div class="book-store">
        <div class="store-text">
          <p class="book-title">"${book.title}"</p>
          <p class="by-text">by</p>
          <p class="book-author">${book.author}</p>
        </div>
        <button class="remove-button" data-index="${index}">Remove</button>
      </div>`);
    this.bookShelf.innerHTML = displayBooks.join('');

    const removeButtons = document.querySelectorAll('.remove-button');
    removeButtons.forEach((btn) => {
      btn.addEventListener('click', () => {
        const { index } = btn.dataset;
        this.removeBook(index);
      });
    });
  }
```

File: modules/index.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  const dt = DateTime.now();
+  return dt.toLocaleString(DateTime.DATETIME_MED_WITH_SECONDS);
+};
+const time = document.querySelector('.date-info');
+setInterval(() => {
+  time.innerHTML = getTime();
+}, 1000);
+/* eslint-disable no-use-before-define */
+/* eslint-disable max-classes-per-file */
+/* eslint-disable-line no-unused-vars */
```
Comment (levy002) [inline modules/index.js]: - [ ] It's crucial not to disable the linter as it plays a pivotal role in achieving better code quality. Linters are designed to catch errors, enforce coding standards, and enhance overall code consistency. By embracing the linter and addressing its suggestions, we can elevate our codebase to higher standards, resulting in improved readability, maintainability, and overall code quality. Kindly do not disable linters.

File: modules/index.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  logo.innerHTML = 'Add Book';
+});
+
+contact.addEventListener('click', () => {
+  section1.style.display = 'none';
+  section2.style.display = 'none';
+  section3.style.display = 'block';
+  section3.style.paddingTop = '160px';
+  logo.innerHTML = 'Contact';
+});
```
Comment (levy002) [inline modules/index.js]: - [ ] I wanted to suggest something regarding the project structure. It's important to place the `index.js` file in the root folder as it serves as the entry point for our JavaScript application. By keeping it in the root directory, we adhere to standard conventions and ensure easy accessibility and execution of our code.

**Review by levy002 — CHANGES_REQUESTED** [review]
## Hello @MargaretTechworld   , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!



- ## Highlights
-  No linter errors :heavy_check_mark: 
- Your app is working well as expected :heavy_check_mark: 

## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by shahadat3669 — APPROVED** [review]
Hi @MargaretTechworld , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ Good Use of GitHub Flows
✅ No linter errors 👍 
✅ Cleane code

## Optional suggestions

- [ ] [Optioanl] Well! Your Readme file looks good. But I notice that your Authors and Future Feature section title are not exist or have proper headings. please follow the [Readme Template](https://github.com/microverseinc/readme-template) and update your Readme file according to it. It will improve your readme file structure. 👍 
image
- [ ]  [Optional] One more thing, after the Acknowledgments section there is two back-to-top links. Please consider removing one to make it looks meaningful.
image

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**